py_gtfs_rt_ingestion/batch_files.py

```diff
@@ -126,20 +122,19 @@ def lambda_handler(event: LambdaDict, context: LambdaContext) -> None:
 
     batch files should all be of same ConfigType
     """
-    logging.info("Processing event:\n%s", json.dumps(event, indent=2))
-    logging.info("Context:\n%s", context)
+    logging.info("batch_event=%s", json.dumps(event))
+    process_logger = ProcessLogger("batch_files_lambda")
+    process_logger.log_start()
 
     try:
         if len(set(event) - set(BatchArgs()._fields)) == 0:
             batch_args = BatchArgs(**event)
         else:
             batch_args = BatchArgs()
-        logging.info(batch_args)
         main(batch_args)
-    except Exception as e:
-        # log if something goes wrong and let lambda recatch the exception
-        logging.exception(e)
-        raise e
+        process_logger.log_complete()
+    except Exception as exception:
+        process_logger.log_failure(exception)
```

Comment:
This would probably be a matter of how we want to parse these splunk logs, but another way to handle completion/failure is to always log completion no matter what, but have a matadata flag available for setting Failed=True/False

From a metrics perspective, might be easier to always assume we will get a completion log no matter what and be able to filter them by `Failed` True/False

Reply:
i had that thought, but ultimately decided that completions and failures are different enough that most metrics don't make sense if they are treated as the same, and we would always be filtering things out based on that boolean. further, it would be something thats easy to forget about and could mess up your metrics (i.e. avg completion time is artificially lower because failed processes take less time to run).